internal/credential/vault/repository_credential_store.go

```diff
@@ -680,15 +680,23 @@ func (r *Repository) UpdateCredentialStore(ctx context.Context, cs *CredentialSt
 				if err != nil {
 					return errors.Wrap(err, op, errors.WithMsg("unable to update credential store version"))
 				}
-				if rowsUpdated != 1 {
+				switch rowsUpdated {
+				case 1:
+				case 0:
+					return nil
+				default:
 					return errors.New(errors.MultipleRecords, op, fmt.Sprintf("updated credential store version and %d rows updated", rowsUpdated))
 				}
 			default:
 				rowsUpdated, err = w.Update(ctx, returnedCredentialStore, filteredDbMask, filteredNullFields, db.NewOplogMsg(&csOplogMsg), db.WithVersion(&version))
 				if err != nil {
 					return errors.Wrap(err, op, errors.WithMsg("unable to update credential store"))
 				}
-				if rowsUpdated != 1 {
+				switch rowsUpdated {
+				case 1:
+				case 0:
+					return nil
```

Comment:
This will be `0` if the store was deleted in a racing request or a version mismatch, should this really return nil here?

Reply:
Yes.  This is how all other resources work with versioned changes.  If the version mismatches or the resource cannot be found it returns nil, no rows updated,  and no error.

Reply:
👍